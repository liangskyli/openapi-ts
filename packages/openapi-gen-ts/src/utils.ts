import type { PathItemObject } from 'openapi-typescript';
import type { Definition } from 'typescript-json-schema';

export const packageName = '@liangskyli/openapi-gen-ts';

export const fileTip = `/**
 * This file was auto-generated by ${packageName}.
 * Do not make direct changes to the file.
 */

`;

export type OpenapiMethod = keyof Omit<
  PathItemObject,
  'servers' | 'parameters' | `x-${string}`
>;
export const methodList: OpenapiMethod[] = [
  'get',
  'put',
  'post',
  'delete',
  'options',
  'head',
  'patch',
  'trace',
];
export type OpenapiDefinition = Omit<Definition, 'properties'> & {
  properties?: { [key: string]: OpenapiDefinition };
};
