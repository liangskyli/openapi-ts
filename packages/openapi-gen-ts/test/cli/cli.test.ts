import { winPath } from '@liangskyli/utils';
import { execa } from 'execa';
import { URL } from 'node:url';
import { describe, expect, test } from 'vitest';

const cwd = new URL('../../', import.meta.url);
const cmd = 'node';
describe('CLI', () => {
  test('CLI test', async () => {
    const { stdout } = await execa(
      cmd,
      ['./bin/index.js', '-c', './test/cli/request.config.ts'],
      {
        cwd,
      },
    );
    await expect(winPath(stdout)).toMatchSnapshot();
  });
});
