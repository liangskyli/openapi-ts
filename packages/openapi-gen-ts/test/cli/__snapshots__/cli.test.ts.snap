// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CLI > CLI test 1`] = `
"[32muse configFile path: ./test/cli/request.config.ts[39m
[32mClean schema-api dir: test/all-gen-dirs/gen-ts-dir-cli1/schema-api[39m
[32mGenerate schema-api/ts-schema.ts success[39m
[32mGenerate schema-api/schema.json success[39m
[32mGenerate schema-api/request.ts file success[39m
[32mGenerate schema-api/interface-api.ts file success[39m
[32mGenerate schema-api/request-api.ts file success[39m
[32mClean schema-api dir: test/all-gen-dirs/gen-ts-dir-cli2/schema-api[39m
[32mGenerate schema-api/ts-schema.ts success[39m
[32mGenerate schema-api/schema.json success[39m
[32mGenerate schema-api/request.ts file success[39m
[32mGenerate schema-api/interface-api.ts file success[39m
[32mGenerate schema-api/request-api.ts file success[39m"
`;
